### PR TITLE
Fix the iCal export for an Event

### PR DESCRIPTION
**Pull request by mic4ael**

### CODE REVIEW

File: indico/MaKaC/webinterface/tpls/EventHeader.tpl [hunk 1]
```diff
@@ -73,11 +73,12 @@ else:
             <%include file="MeetingFilter.tpl"/>
         % endif
         % if showExportToICal:
-            <a id="exportIcal${self_._conf.getUniqueId()}" href="#" class="exportIcal" data-id="${self_._conf.getUniqueId()}">
-                ${ _("iCal export") }
-                <div class="leftCorner"></div>
+            <a id="exportIcal${self_._conf.as_event.id}" href="#" class="exportIcal"
+               data-id="${self_._conf.as_event.id}">
+               ${ _("iCal export") }
```
Comment (nop33) [inline indico/MaKaC/webinterface/tpls/EventHeader.tpl]: You moved the indentation one space to the left 😝

File: indico/modules/events/__init__.py [hunk 2]
```diff
@@ -281,3 +281,11 @@ def _render_event_person_link_list_field(event, event_type, **kwargs):
     form = EventPersonLinkForm(event=event, event_type=event_type)
     tpl = get_template_module('forms/_form.html')
     return tpl.form_row(form.person_link_data, skip_label=True)
+
+
+@template_hook('event-ical-export')
+def _render_event_ical_export(event, **kwargs):
+    from indico.modules.events.util import get_base_ical_parameters
```
Comment (nop33) [inline indico/modules/events/__init__.py]: Hm.. There was a boy one day (yesterday) that said:
> I really don't understand people who don't leave an empty line between local imports and the code.

File: indico/modules/events/blueprint.py [hunk 3]
```diff
@@ -16,6 +16,7 @@
 
 from __future__ import unicode_literals
 
+from indico.modules.events.controllers.display import RHExportEventICAL
```
Comment (nop33) [inline indico/modules/events/blueprint.py]: You gotta move it one line below

File: indico/modules/events/__init__.py (hunk 2 above)
Comment (OmeGak) [inline indico/modules/events/__init__.py]: But we all know this boy is on a journey for self improvement. I see he's learning good manners.

Comment (mic4ael) [inline indico/modules/events/__init__.py]: He told me that even though he has his own preferred style, he still needs to follow the rules in the team.

Comment (nop33) [inline indico/modules/events/__init__.py]: DON'T LOSE YOUR IDENTITY!

Comment (ThiefMaster) [inline indico/modules/events/__init__.py]: I don't think we have a rule for whitespace after local imports ;)

File: indico/modules/events/sessions/util.py [hunk 4]
```diff
@@ -156,23 +157,18 @@ def get_sessions_for_user(event, user):
...
-        '_fossil': 'sessionMetadata',
+    from indico.modules.events.contributions.util import serialize_contribution_for_ical
+    from indico.modules.events.util import serialize_person_link
+    return {
+        '_fossil': 'sessionMetadataWithContributions',
         'id': sess.id,
         'startDate': sess.start_dt,
         'endDate': sess.end_dt,
-        'url': '',
+        'url': url_for('sessions.display_session', sess, _external=True),
```
Comment (nop33) [inline indico/modules/events/sessions/util.py]: Why do you use `unicode` in L245 of `indico/modules/events/contributions/util.py` and here not? Is it really necessary btw?

File: indico/modules/events/contributions/util.py [hunk 5]
```diff
@@ -237,26 +237,19 @@ def get_contributions_with_user_as_submitter(event, user):
...
 
 def serialize_contribution_for_ical(contrib):
-    contrib_data = {
+    return {
         '_fossil': 'contributionMetadata',
         'id': contrib.id,
         'startDate': contrib.timetable_entry.start_dt if contrib.timetable_entry else None,
         'endDate': contrib.timetable_entry.end_dt if contrib.timetable_entry else None,
-        'url': url_for('contributions.display_contribution', contrib, _external=True),
+        'url': to_unicode(url_for('contributions.display_contribution', contrib, _external=True)),
```
Comment (ThiefMaster) [inline indico/modules/events/contributions/util.py]: it's already unicode

File: indico/modules/events/sessions/util.py (hunk 4 above)
Comment (mic4ael) [inline indico/modules/events/sessions/util.py]: I think I forgot to remove the call to `to_unicode`

**Pull request merged**